src/FxCop/Desktop.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithmsAnalyzer.cs

```diff
@@ -130,30 +153,33 @@ public void AnalyzeNode(SyntaxNodeAnalysisContext context)
 
                 if (type.IsDerivedFrom(this._cryptTypes.DES, baseTypesOnly: true))
                 {
-                    rule = DoNotUseDESSpecificRule;
+                    rule = CreateCA5351DiagnosticDescriptor(owningParentName, _cryptTypes.DES.Name);
```

Comment:
We shouldn't be allocation another descriptor here. If we get had a field for the for the descriptor then this method goes away. The descriptor shouldn't change based on the parent name - the descriptor applies to the entire class of the diagnsotics and the diagnostic is an instance and so the parentName belongs in the diagnostic. If you reword your message\description the way I have suggested above - this need would go away.